lib/elixir/lib/process.ex

```diff
@@ -421,6 +421,12 @@ defmodule Process do
   @spec register(pid | port, atom) :: true
   def register(pid, name) when not name in [nil, false, true] and is_atom(name) do
     :erlang.register(name, pid)
+  catch
```

Comment:
I have handled these cases in https://github.com/elixir-lang/elixir/commit/16f73ab50dc8b4e09a907971e050768ff10a40d5